distributed/tests/test_failed_workers.py

```diff
@@ -41,30 +41,31 @@ def test_submit_after_failed_worker_sync(loop):
 
 
 @pytest.mark.slow()
-@gen_cluster(client=True, timeout=60, active_rpc_timeout=10)
-async def test_submit_after_failed_worker_async(c, s, a, b):
+@pytest.mark.parametrize("compute_on_failed", [False, True])
+@pytest.mark.parametrize("TEMPrun", range(100))
+@gen_cluster(client=True, config={"distributed.comm.timeouts.connect": "500ms"})
+async def test_submit_after_failed_worker_async(c, s, a, b, compute_on_failed, TEMPrun):
     async with Nanny(s.address, nthreads=2) as n:
-        while len(s.workers) < 3:
-            await asyncio.sleep(0.1)
+        await c.wait_for_workers(3)
 
         L = c.map(inc, range(10))
         await wait(L)
 
-        s.loop.add_callback(n.kill)
-        total = c.submit(sum, L)
-        result = await total
-        assert result == sum(map(inc, range(10)))
+        kill_task = asyncio.create_task(n.kill())
```

Comment:
this test seems very timing sensitive, how far through is the `n.kill()` task and what state is the Nanny in by the time the `Client._gather_remote(` gets called?

Reply:
The test is indeed very timing sensitive, and as I discovered elsewhere it can need to run a good 100+ times to trigger unexpected behaviour. I don't plan to fix this in the scope of this PR. The change is just to clean up the code.